airflow/models.py

```diff
@@ -1288,19 +1288,19 @@ def run(
             verbose=True)
 
         if not runnable and not mark_success:
-            if self.state != State.QUEUED:
-                # If a task's dependencies are met but it can't be run yet then queue it
-                # instead
-                self.state = State.QUEUED
-                msg = "Queuing attempt {attempt} of {total}".format(
-                    attempt=self.try_number % (task.retries + 1) + 1,
-                    total=task.retries + 1)
-                logging.info(hr + msg + hr)
-
-                self.queued_dttm = datetime.now()
-                msg = "Queuing into pool {}".format(self.pool)
-                logging.info(msg)
-                session.merge(self)
+            # we might have hit concurrency limits, which means we probably
+            # have been running prematurely.
+            self.state = State.UP_FOR_RETRY
```

Comment:
@saguziel No, if the TI sets the status to QUEUED it will not get executed. Doing this resulted in arbitrary tasks getting queued twice (or more) in the executor. It is the reason why I feel quite strongly about not allowing a task instance to "dump" itself right in the middle of what the scheduler should handle.